analysis/quantifier-comparisons/09-emptydrop-comparisons.Rmd

```diff
@@ -0,0 +1,461 @@
+}
+
+knee_filtered_sces <- fry_sces %>%
+  purrr::map(knee_filter)
+
+readr::write_rds(knee_filtered_sces, file.path(results_dir, "alevin-fry-cr-like-em-knee-filtering-sces.rds"))
+```
+
+
+## Compare Cells between EmptyDrops and cellRangerLikeEmptyDrops
+
+Now that we have performed the different filtering strategies we are ready to compare the cells that remain after filtering. 
+We will first take a look at the number of cells that remain post filtering and compare those numbers to the Cell Ranger results. 
+Then we will look at the distribution of UMI/cell and genes/cell with different filtering strategies in comparison to Cell Ranger. 
+
+### Number of Cells Filtered
+
+```{r}
+# function to grab the number of cells from an sce object 
+get_num_cells <- function(sce){
+  num_cells <- dim(sce)[2]
+}
+```
+
+
+```{r}
+emptydrops_cell_num <- emptydrops_filter_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(emptydrops_cell_num) <- c("emptyDrops")
+
+emptydrops_200_cell_num <- emptydrops_filter_sces_200 %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(emptydrops_200_cell_num) <- c("emptyDrops200")
+
+emptydrops_500_cell_num <- emptydrops_filter_sces_500 %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(emptydrops_500_cell_num) <- c("emptyDrops500")
+
+cr_filter_cell_num <- crlike_filter_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(cr_filter_cell_num) <- c("emptyDropsCellRanger")
+
+knee_filter_cell_num <- knee_filtered_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame() 
+rownames(knee_filter_cell_num) <- c("knee")
+
+names(cellranger_sces) <- sample_names[3:6]
+# add additional nuclei samples from previous benchmarking 
+cellranger_sces[["SCPCR000118-nuclei"]] <- cellranger_nuclei_sces[["SCPCR000118-cdnapre_mRNA"]]
+cellranger_sces[["SCPCR000119-nuclei"]] <- cellranger_nuclei_sces[["SCPCR000119-cdnapre_mRNA"]]
+                                                                  
+cellranger_cell_num <- cellranger_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(cellranger_cell_num) <- c("CellRanger")
+
+# create combined dataframe with number of cells and filtering method 
+combined_cell_num_df <- dplyr::bind_rows(emptydrops_cell_num,
+                                         emptydrops_200_cell_num,
+                                         cr_filter_cell_num, 
+                                         knee_filter_cell_num,
+                                         cellranger_cell_num,
+                                         emptydrops_500_cell_num) %>%
+  tibble::rownames_to_column("filtering_method") %>%
+  tidyr::pivot_longer(cols = starts_with("SCPCR"),
+                      names_to = "sample", 
+                      values_to = "number_cells")
+```
+
+
+```{r fig.height=7, fig.width=10}
+ggplot(combined_cell_num_df, aes(x = sample, y = number_cells, color = filtering_method)) + 
+  geom_jitter() +
+  geom_sina(size = 4) + 
```

Comment:
Do we need jitter and sina here? I'm not sure we want either, but maybe a very small jitter? I found it very hard to read this plot in its current form. This might actually be a place for a bar chart (crazy talk).
```suggestion
  geom_col(position = "dodge") +
```